pyproject.toml

```diff
@@ -1,3 +1,110 @@
 [build-system]
 requires = ["setuptools >= 66.1.0"]  # minimum setuptools version supporting Python 3.12
 build-backend = "setuptools.build_meta"
+
+[project]
+requires-python = ">=3.10"
+name = "ansible-core"
+authors = [
+    {name = "Ansible Project"},
+]
+description = "Radically simple IT automation"
+readme = "README.md"
+license = {text = "GPLv3+"}
+classifiers = [
+    "Development Status :: 5 - Production/Stable",
+    "Environment :: Console",
+    "Intended Audience :: Developers",
+    "Intended Audience :: Information Technology",
+    "Intended Audience :: System Administrators",
+    "License :: OSI Approved :: GNU General Public License v3 or later (GPLv3+)",
+    "Natural Language :: English",
+    "Operating System :: POSIX",
+    "Programming Language :: Python :: 3",
+    "Programming Language :: Python :: 3.10",
+    "Programming Language :: Python :: 3.11",
+    "Programming Language :: Python :: 3 :: Only",
+    "Topic :: System :: Installation/Setup",
+    "Topic :: System :: Systems Administration",
+    "Topic :: Utilities",
+]
+dynamic = ["version", "dependencies"]
+
+[project.urls]
+"Homepage" = "https://ansible.com/"
+"Bug Tracker" = "https://github.com/ansible/ansible/issues"
+"CI: Azure Pipelines" = "https://dev.azure.com/ansible/ansible/"
+"Code of Conduct" = "https://docs.ansible.com/ansible/latest/community/code_of_conduct.html"
+"Documentation" = "https://docs.ansible.com/ansible-core/"
+"Mailing lists" = "https://docs.ansible.com/ansible/latest/community/communication.html#mailing-list-information"
+"Source Code" = "https://github.com/ansible/ansible"
+
+[tool.setuptools.dynamic]
+version = {attr = "ansible.release.__version__"}
+dependencies = {file = "requirements.txt"}
+
+[tool.setuptools]
+zip-safe = false
```

Comment:
Is `zip-safe` actually needed? We're not building eggs.

Reply:
I didn't check to see if it was necessary. I just copied it over from the old `setup.cfg`:

https://github.com/ansible/ansible/blob/f10d11bcdc54c9b7edc0111eb38c59a88e396d0a/setup.cfg#L38

Reply:
It wasn't generating tracebacks if built in the normal way, it just produced warnings.

Anyway, a lot of time has passed and that flag is now considered obsolete since egg and easy_install were deprecated. I expect it to make no difference right now.

Reply:
I've omitted `zip_safe = False` in the latest rebase.